nm dns: Store DNS to interface also when global DNS been used

cathay4t opened the pull request:
https://github.com/coreos/fedora-coreos-tracker/issues/1947 indicate
OpenShift assisted installer depend on NM keyfile for installation from
PXE image where only NM keyfile can be preserved after switching from
minimum ramfs to full ramfs and finally to on-disk OS.

To workaround that issue, also store DNS information into desired interface even
when global DNS been used.

Integrator test case included and marked as tier1.

Resolves: https://issues.redhat.com/browse/RHEL-91250

cathay4t commented:
For NetworkManager part, we need fix in https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/merge_requests/2210

ihuguet commented on rust/src/lib/nm/dns.rs:
```diff
@@ -124,12 +126,15 @@ fn find_dns_iface(
     cur_dns_ifaces: &[String],
     nm_acs: &[NmActiveConnection],
     nm_devs: &[NmDevice],
+    desired_only: bool,
```
What is this for? AFAIK, all the callers pass `false`, so this argument seems useless.

cathay4t replied:
Fixed. `store_dns_config_to_desired_iface()` should set it true.

cathay4t committed:
nm dns: Store DNS to interface also when global DNS been used

https://github.com/coreos/fedora-coreos-tracker/issues/1947 indicate
OpenShift assisted installer depend on NM keyfile for installation from
PXE image where only NM keyfile can be preserved after switching from
minimum ramfs to full ramfs and finally to on-disk OS.

To workaround that issue, also store DNS information into desired
interface even when global DNS been used.

Integrator test case included and marked as tier1.

Resolves: https://issues.redhat.com/browse/RHEL-91250

Signed-off-by: Gris Ge <fge@redhat.com>
